Provide baseline access control to web services

Enet4 opened the pull request:
Should resolve #98 for Dicoogle core services. Security of services in extensions still needs to be upheld by each plugin.

### Summary

- Add another mechanism to stay logged in using a cookie
   - `DICOOGLE_SESSION` is automatically provided on login and is looked up when the `Authorization` header is not found.
- Require to be logged in for most core Dicoogle services
   - Web service extensions via plugins are not included here
   - Require user logged in to be an admin in management related operations
- Add server setting "allow-unauthorized"
   - If `true`, access control is disabled like in dicoogle<=3.5.0
   - A warning is displayed when starting the web server with this flag enabled
- Extend Dicoogle platform interface so that plugins can know that a user is authenticated

### Pending issues

- [x] Requests to `webui/module` and `/dic2png` (for the thumbnails) in the webapp do not have the authentication token at the moment, so they will always fail.
   - Resolved by setting a session cookie.
- [x] Errors in user presets, likely a regression in another PR.
   - Fixed in #738

Enet4 committed:
Provide baseline access control to web services

- require to be logged in for most core Dicoogle services
   - Web service extensions via plugins are not included here
   - Require user logged in to be an admin in management related operations
- require and admin account for
- add server setting "allow-unauthorized"
   - if `true`, access control is disabled like in dicoogle<=3.5.0
   - a warning is displayed when starting the web server with this flag enabled

Enet4 committed:
Fix authorization in DicoogleWeb webui module handler

- check whether authorization is enabled before adding the auth filter

Enet4 committed:
Extends the ways Dicoogle server finds a Dicoogle user session token

- allow tokens in Authorization header prefixed by "Dicoogle " or "Bearer "
- If not found, look for token in cookie named DICOOGLE_SESSION
- Add `Authentication#getAuthenticatedUser` to easily fetch the user object associated with the token
- Update other endpoints to adopt the right authenticated user retrieval practices

Enet4 committed:
Fix LoginServlet get method

- Fetch token from request and retrieve user manually,
  as this servlet is not filtered by the authentication filter

Enet4 committed:
Revise `Authentication.getAuthenticatedUser`

- Make method also work even without the authentication filter.
- Update documentation accordingly
- Set HttpOnly to Dicoogle session cookie

tiberio-baptista reviewed (approved):
Tested locally.

After a few issues with local maven dependencies and cache, I was able to validate the following behaviors:

* Usage of web UI:
  * Log in and be able to use management features with an admin account
  * Log out in other tab, confirm that former tab becomes unable to do anything until new login session is issued
* Usage of web API
  * Confirmed authentication is required for managerial Dicoogle API requests (enable/disable storage, list/create users)
  * Confirmed that valid DICOOGLE_SESSION cookie allows API usaeg if `Authorization` header is absent
* Confirm `allow-unauthorized` disables these auth features, and warning is issued in Dicoogle logs.

I believe everything is in good order, so I'm approving this PR.

Enet4 committed:
Extend dicoogle platform with `getUser` methods

- add simple POJO DicoogleUser
   - same as `User` in core, but without a password hash
- add `getUser(Request)` and `getUser(String token)` to `DicooglePlatformInterface`
   - allows plugins to implement authorization based on dicoogle user sessions more easily
- Update dicoogle platform implementations accordingly
